OpenRA.Mods.Common/Traits/Explodes.cs

```diff
@@ -41,7 +41,7 @@ public void Killed(Actor self, AttackInfo e)
 			if (self.World.SharedRandom.Next(100) > explodesInfo.Chance)
 				return;
 
-			if (explodesInfo.DeathType != null && e.Warhead != null && !explodesInfo.DeathType.Contains(e.Warhead.DeathType))
+			if (explodesInfo.DeathType != null && e.Warhead != null && !explodesInfo.DeathType.Any(x => e.Warhead.DamageTypes.Contains(x)))
```

Comment:
No, it doesn't set a bad example.
We use `Intereset()` in many places.
Do you have any actual numbers to prove that it is less efficient?